src/entries-list/entries-list.ts

```diff
@@ -0,0 +1,178 @@
+import { LitElement, html, css } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import {
+  getDashboardEntries,
+  ImportableEntry,
+  ListEntriesResult,
+} from "../api/list-entries";
+import { openWizardDialog } from "../wizard";
+import "@material/mwc-button";
+import "../components/esphome-button-menu";
+import "../components/esphome-card";
+import { subscribeOnlineStatus } from "../online_status";
+import "./node-card";
+import "./import-card";
+
+@customElement("esphome-entries-list")
+class ESPHomeEntriesList extends LitElement {
+  @state() private _entries?: ListEntriesResult;
+  @state() private _onlineStatus?: Record<string, boolean>;
+  @state() private _highlightedName?: string;
+
+  private _updateEntriesInterval?: number;
+  private _nodeStatusUnsub?: () => void;
+
+  protected render() {
+    if (this._entries === undefined) {
+      return html``;
+    }
+    if (
+      this._entries.configured.length === 0 &&
+      this._entries.importable.length === 0
+    ) {
+      return html`
+        <div class="welcome-container">
+          <h5>Welcome to ESPHome</h5>
+          <p>It looks like you don't yet have any nodes.</p>
+          <p>
+            <mwc-button
+              raised
+              label="Add node"
+              icon="add"
+              @click=${openWizardDialog}
+            ></mwc-button>
+          </p>
+        </div>
+      `;
+    }
+
+    const importable = this._entries.importable;
+
+    return html`
+      <div class="container">
+        ${importable.length
+          ? html`
+        <h5>Discovered ESPHome devices</h5>
+        <div class="grid">
+          ${importable.map(
+            (entry) => html`
+              <esphome-import-card
+                .entry=${entry}
+                @imported=${() => this._handleImported(entry)}
+              >
+              </esphome-import-card>
+            `
+          )}
+        </div>
+        <hr></hr>
+      `
+          : ""}
+
+        <div class="grid">
+          ${this._entries.configured.map(
+            (entry) => html` <esphome-node-card
```

Comment:
If you put the HTML opening and closing tag on different lines than the string opening/closing, it formats nicer.